src/otio_aaf_adapter/adapters/advanced_authoring_format.py

```diff
@@ -841,173 +992,43 @@ def _transcribe(item, parents, edit_rate, indent=0):
     # There's a bit more we can do before we're ready to return the result.
 
     # If we didn't get a name yet, use the one we have in metadata
-    if not result.name:
+    if not result.name and "Name" in metadata:
         result.name = metadata["Name"]
 
     # Attach the AAF metadata
     if not result.metadata:
         result.metadata.clear()
-    result.metadata["AAF"] = metadata
+
+    if "AAF" not in result.metadata:
+        result.metadata["AAF"] = metadata
 
     # Double check that we got the length we expected
     if isinstance(result, otio.core.Item):
         length = metadata.get("Length")
         if (
-                length
-                and result.source_range is not None
-                and result.source_range.duration.value != length
+                length and
+                result.source_range is not None and
+                result.source_range.duration.value != length
         ):
             raise AAFAdapterError(
-                "Wrong duration? {} should be {} in {}".format(
+                "Wrong duration? {} should be {} {} {} in {}".format(
```

Comment:
Since we're all Python 3 now, we could use f-strings for these, make it more legible.